distributed/shuffle/tests/test_rechunk.py

```diff
@@ -204,6 +204,35 @@ async def test_rechunk_configuration(c, s, *ws, config_value, keyword):
     assert np.all(await c.compute(x2) == a)
 
 
+@pytest.mark.parametrize(
+    ["new", "expected_algorithm"],
+    [
+        # All-to-all rechunking defaults to P2P
+        (((1,) * 100, (100,)), "p2p"),
+        # Localized rechunking defaults to tasks
+        (((50, 50), (2,) * 50), "tasks"),
+        # Less local rechunking first defaults to tasks,
+        (((25, 25, 25, 25), (4,) * 25), "tasks"),
+        # then switches to p2p
+        (((10,) * 10, (10,) * 10), "p2p"),
+    ],
+)
+@gen_cluster(client=True)
+async def test_rechunk_heuristic(c, s, a, b, new, expected_algorithm):
```

Comment:
I'm a little surprised we're testing this here instead of over in `dask/dask`, where the definition of the heuristic lives. I don't mean this to be a blocking comment -- I'm glad there's test coverage for this somewhere : )

I'm a little out of date on how we're choosing where rechunking-related tests go. Why did you put this test here instead of in https://github.com/dask/dask/pull/11337?

Reply:
All existing tests (that I'm aware of) around rechunk configuration live in `test_rechunk.py`. I don't think there's a perfect place to put these as these two components are generally intertwined and P2P requires a distributed cluster. I could see two things happening:

* Move the logic around choosing between P2P and tasks on a distributed cluster over into `distributed`. This way, it lives in the same repo as the tests.
* Add a test to `dask` that simply tests that we use `tasks`-based rechunking when not using a distributed cluster.

Reply:
Ah, I see. Agreed that this is sort of a gray area where there's not a clear and obvious place for these to live. What you have here is fine and we can always move things around later if we want.